### ISSUE TITLE
Not able to query single collection on GraphQL with anything other than ID

### ISSUE THREAD

**Issue by creative-andrew** [payloadcms/payload#3148]
### Link to reproduction

https://www.wpgraphql.com/docs/posts-and-pages#single-post-by-slug

### To Reproduce

It is a common practice to use `slugs` or something similar to identify the collection rather than the `ID`. 
Right now you can only pass the `id` to GraphQL to get a single collection document.
Hence, you cannot create a `GetPageBySlug` query. 

![image](https://github.com/payloadcms/payload/assets/18492002/398ff95f-d8e7-4e9c-a181-62c2b58613c5)

Payload provides [a way to declare a custom ID](https://payloadcms.com/docs/fields/overview#customizable-id).
However, this won't work in more than one language, as you cannot localize a custom id.

The idea here is to be able to declare a field to be used in GraphQL as a unique identifier other than the ID.
Collections support having unique fields, which would work really nicely with this as they are unique within the collection. However, I am not sure if they are also unique also on each language. 

```ts
export const Pages: CollectionConfig = {
  slug: "pages",
  fields: [
    {
      name: "slug",
      type: "text",
      required: true,
      localized: true,
     unique: true,
     useAsIdentifier: true
    },
};
```
```graphql
{
  Page(id: "mySlug" idType: SLUG)
}
```

`idType` is an enum that by default includes `Id` and other unique identifiers defined in a field. 

Here is an example from WordPress WPGraphQL solving this issue:
https://www.wpgraphql.com/docs/posts-and-pages#single-post-by-slug

### Describe the Bug

Not able to query single collection on GraphQL with anything other than ID

### Payload Version

1.9.0

**Comment by JarrodMFlesch** [comment]
Do you need the slug to be localized? Is that common practice?

**Comment by creative-andrew** [comment]
That is in fact a common practice.
Imagine you have a page with a slug "example-api-development-payload" for EN.
The translation of that page should have its own slug "ejemplo-api-desarollo-payload"
This is important for SEO.

Here are some webs with that behavior:
https://www.hauraton.com/en/products/
Please use the language switcher to see the slug changing.

**Comment by creative-andrew** [comment]
> You could use a where query to fetch a single doc though. Yes it will come back in an array of docs, but you will only have 1 document inside. That is normally what we do when we need to fetch documents by slugs, which I know is a common thing.

Yes, what you suggested is what we ended up doing! However, I think we should leave the issue open until we found a more permanent solution.

**Comment by JarrodMFlesch** [comment]
@creative-andrew, I don't know. I feel like this is a feature request and not an issue? Maybe we should convert it to a discussion. I see the value in it, I just am not sure we will keep it open as an "issue".

**Comment by creative-andrew** [comment]
Sure thing. Let's do that.
